DigitalLearningSolutions.Web.Tests/Controllers/VerifyEmailControllerTests.cs

```diff
@@ -144,5 +146,33 @@ public void Index_with_unverified_email_matching_code_updates_records_and_return
 
             result.Should().BeViewResult().WithDefaultViewName();
         }
+
+        [Test]
+        public void Index_with_unverified_email_for_unapproved_delegate_passes_centre_id_to_the_view()
+        {
+            // Given
+            const int userId = 1;
+            var now = new DateTime(2022, 1, 1);
+
+            var emailVerificationDetails = new EmailVerificationDetails
+            {
+                UserId = userId,
+                Email = Email,
+                EmailVerificationHash = Code,
+                EmailVerified = null,
+                EmailVerificationHashCreatedDate = now,
+                CentreIdIfEmailIsForUnapprovedDelegate = 2,
+            };
+
+            A.CallTo(() => clockUtility.UtcNow).Returns(now);
+            A.CallTo(() => userService.GetEmailVerificationDetails(Email, Code)).Returns(emailVerificationDetails);
+            A.CallTo(() => userService.SetEmailVerified(userId, Email, now)).DoesNothing();
+
+            // When
+            var result = controller.Index(Email, Code);
+
+            // Then
+            result.Should().BeViewResult().ModelAs<int>();

```

Comment:
Think you can check the value of the model as well.

Reply:
done!